### PR TITLE
build(deps): bump Band Protocol to v2.3.1

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description

Bump Band Protocol dependencies to version v2.3.1, which is going to be the Band version used inside their new testnet as well as upcoming mainnet. This version relies on Cosmos v0.44.0. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [x] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [x] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/desmos/profiles/v1beta1/models_app_links.proto [hunk 1]
```diff
@@ -59,11 +59,11 @@ message OracleRequest {
   option (gogoproto.equal) = true;
 
   // ID is the ID of the request
-  int64 id = 1
+  uint64 id = 1
```
Comment (RiccardoM) [inline proto/desmos/profiles/v1beta1/models_app_links.proto:62]: All these `int` are now `uint` inside Band code

File: proto/desmos/profiles/v1beta1/models_params.proto [hunk 2]
```diff
@@ -107,13 +107,9 @@ message OracleParams {
   // the oracle script
   uint64 execute_gas = 5 [ (gogoproto.moretags) = "yaml:\"execute_gas\"" ];
 
-  // FeePayer represents the key of the account that is going to pay for oracle
-  // fees if needed
-  string fee_payer = 6 [ (gogoproto.moretags) = "yaml:\"fee_payer\"" ];
```
Comment (RiccardoM) [inline proto/desmos/profiles/v1beta1/models_params.proto:112]: This is no longer used by Band Protocol

File: x/profiles/client/cli/cli_app_links_test.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func (s *IntegrationTestSuite) TestCmdQueryApplicationsLinks() {
 						types.NewData("reddit", "reddit-user"),
 						types.ApplicationLinkStateInitialized,
 						types.NewOracleRequest(
-							-1,
+							0,
```
Comment (RiccardoM) [inline x/profiles/client/cli/cli_app_links_test.go:38]: `uint` does not support negative values, so the default one will be `0` now

File: .changeset/entries/397a12a6296b6f866be6f048c1be9e500e8a3a570c96b1dec4b1cd451ae142f9.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+type: deps
+module: other
+pull_request: 619
+description: Upodated Band Protocol to v2.3.1
```
Comment (bragaz) [inline .changeset/entries/397a12a6296b6f866be6f048c1be9e500e8a3a570c96b1dec4b1cd451ae142f9.yaml:4]:
```suggestion
description: Updated Band Protocol to v2.3.1
```

File: x/oracle/types/error.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+	ErrRequestNotFound          = sdkerrors.Register(ModuleName, 5, "request not found")
+	ErrRawRequestNotFound       = sdkerrors.Register(ModuleName, 6, "raw request not found")
+	ErrReporterNotFound         = sdkerrors.Register(ModuleName, 7, "reporter not found")
+	ErrResultNotFound           = sdkerrors.Register(ModuleName, 8, "result not found")
+	ErrReporterAlreadyExists    = sdkerrors.Register(ModuleName, 9, "reporter already exists")
+	ErrValidatorNotRequested    = sdkerrors.Register(ModuleName, 10, "validator not requested")
+	ErrValidatorAlreadyReported = sdkerrors.Register(ModuleName, 11, "validator already reported")
+	ErrInvalidReportSize        = sdkerrors.Register(ModuleName, 12, "invalid report size")
+	ErrReporterNotAuthorized    = sdkerrors.Register(ModuleName, 13, "reporter not authorized")
+	ErrEditorNotAuthorized      = sdkerrors.Register(ModuleName, 14, "editor not authorized")
```
Comment (bragaz) [inline x/oracle/types/error.go:21]: missing the 15 or is It ok?

File: x/oracle/types/packets.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewOracleRequestPacketData contructs a new OracleRequestPacketData instance
```
Comment (bragaz) [inline x/oracle/types/packets.go:7]:
```suggestion
// NewOracleRequestPacketData constructs a new OracleRequestPacketData instance
```

File: x/oracle/types/error.go (hunk 5 above)
Comment (RiccardoM) [reply x/oracle/types/error.go:21]: @bragaz These files are automatically downloaded from the Band Protocol repository, so I think it's fine. We only use part of them anyway inside our codebase

**Review by bragaz — APPROVED** [review]

**Pull request merged**